pkg/resourcecollector/persistentvolumeclaim.go

```diff
@@ -96,6 +98,26 @@ func (r *ResourceCollector) preparePVCResourceForApply(
 		}
 
 	}
+	if len(storageClassMappings) > 0 {
+		// In the case of storageClassMappings, we need to reset the
+		// storage class annotation and the provisioner annotation
+		var newSc string
+		var exists bool
+		if val, ok := pvc.Annotations[utils.StorageClassAnnotationKey]; ok {
+			if newSc, exists = storageClassMappings[val]; exists && len(newSc) > 0 {
+				pvc.Annotations[utils.StorageClassAnnotationKey] = newSc
+			}
+		}
+		if _, ok := pvc.Annotations[utils.StorageProvisionerAnnotationKey]; ok {
+			if len(newSc) > 0 {
+				storageClass, err := storage.Instance().GetStorageClass(newSc)
```

Comment:
since this is in `resourcecollector` pkg, lets use passed in client for storageclass instead of incluster config

Reply:
Fixed